src/main/java/org/apache/mesos/offer/DynamicPortRequirement.java

```diff
@@ -0,0 +1,94 @@
+package org.apache.mesos.offer;
+
+import org.apache.mesos.Protos;
+
+/**
+ * Created by gabriel on 9/28/16.
+ */
+public class DynamicPortRequirement extends ResourceRequirement {
+    private static final String DYNAMIC_PORT_KEY = "dynamic_port_key";
```

Comment:
Are resource labels used elsewhere? Feels like we'll want to have a central listing of label keys soon, if there isn't one already.